communities.go

```diff
@@ -0,0 +1,197 @@
+		if err != nil {
+			return nil, nil, err
+		}
+		censusChannel = &Channel{
+			ID:          channel.ID,
+			Name:        channel.Name,
+			Description: channel.Description,
+			Followers:   channel.Followers,
+			ImageURL:    channel.Image,
+			URL:         channel.URL,
+		}
+	} else { // ERC20 or NFT census type
+		censusAddresses = []*CensusAddress{}
+		if len(dbCensus.Addresses) > 0 {
+			for _, addr := range dbCensus.Addresses {
+				censusAddresses = append(censusAddresses, &CensusAddress{
+					Address:    addr.Address,
+					Blockchain: addr.Blockchain,
+				})
+			}
+		}
+	}
+	return censusAddresses, censusChannel, nil
+}
+
+func (v *vocdoniHandler) listCommunitiesHandler(msg *apirest.APIdata, ctx *httprouter.HTTPContext) error {
+	var err error
+	var dbCommunities []mongo.Community
+	// check if the query has the byAdminFID or byAdminUsername parameters and
+	// list communities by admin FID or username respectively, otherwise list
+	// all communities
+	byAdminFID := ctx.Request.URL.Query().Get("byAdminFID")
+	byAdminUsername := ctx.Request.URL.Query().Get("byAdminUsername")
+	switch {
+	case byAdminFID != "":
+		// if the query has the byAdminFID parameter, list communities by admin FID
+		var adminFID int
+		adminFID, err = strconv.Atoi(byAdminFID)
+		if err != nil {
+			return ctx.Send([]byte("Invalid admin FID"), http.StatusBadRequest)
+		}
+		if dbCommunities, err = v.db.ListCommunitiesByAdminFID(uint64(adminFID)); err != nil {
+			return ctx.Send([]byte("Error listing communities"), http.StatusInternalServerError)
+		}
+	case byAdminUsername != "":
+		// if the query has the byAdminUsername parameter, list communities by admin username
+		if dbCommunities, err = v.db.ListCommunitiesByAdminUsername(byAdminUsername); err != nil {
+			return ctx.Send([]byte("Error listing communities"), http.StatusInternalServerError)
+		}
+	default:
+		// otherwise, list all communities
+		if dbCommunities, err = v.db.ListCommunities(); err != nil {
+			return ctx.Send([]byte("Error listing communities"), http.StatusInternalServerError)
+		}
+	}
+	if len(dbCommunities) == 0 {
+		return ctx.Send([]byte("No communities found"), http.StatusNotFound)
+	}
+	communities := CommunityList{
+		Communities: []*Community{},
+	}
+	for _, c := range dbCommunities {
+		// get admin profiles from the database
+		admins := []*User{}
+		for _, admin := range c.Admins {
+			user, err := v.db.User(admin)
+			if err != nil {
+				if err == farcasterapi.ErrNoDataFound {
+					continue
+				}
+				return ctx.Send([]byte(err.Error()), apirest.HTTPstatusInternalErr)
+			}
+			admins = append(admins, &User{
+				FID:         user.UserID,
+				Username:    user.Username,
+				DisplayName: user.Displayname,
+				// Avatar:      user.Avatar,
+			})
+		}
+		// get census channel or addresses based on the type
+		cAddresses, cChannel, err := v.censusChannelOrAddresses(ctx.Request.Context(), c.Census)
+		if err != nil {
+			if err == farcasterapi.ErrChannelNotFound {
```

Comment:
This is weird. Why are you just wrapping the ErrChannelNotFound error (returning 404) while other error are 500? Both errors should be wrapper or none. I'd say not to wrap the error and just return whatever the v.censusChannelOrAddresses() method returns with a 404.